src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,358 @@
+
+        try {
+
+            // Depending on command, we process the command
+            // Exceptions will be thrown respectively
+            switch (commandType) {
+
+            case "help":
+                return Ui.welcome();
+
+            case "bye":
+                return Ui.goodbye();
+
+            case "list":
+                return Ui.listAllTasksMessage(this);
+
+            case "delete":
+                return deleteTask(command);
+
+            case "done":
+                return doneTask(command);
+
+            case "todo":
+                return addNewToDo(command);
+
+            case "deadline":
+                return addNewDeadline(command);
+
+            case "event":
+                return addNewEvent(command);
+
+            case "find":
+                return findTask(command);
+
+            default:
+                throw new InputUnclearException("");
+
+            }
+        } catch (DoneException | DeadlineException | DeleteException | EmptyTaskListException
+            | EventException | InputUnclearException | ToDoException e) {
+            return Ui.exceptionMessage(e.toString());
+        }
+
+    }
+
+    /**
+     * Gives a String representation of the result of completing a selected Task, and marks the Task as done.
+     *
+     * @param command the full, unedited command entered by the user.
+     * @return String representation of a message indicating the selected Task is completed.
+     * @throws DoneException possible Exception when running a done command.
+     * @throws InputUnclearException an Exception given when the input is unfitting.
+     */
+    private String doneTask(String command) throws DoneException, InputUnclearException {
+        // user inputs will be "done _____"
+        // Take only the very next token which must be an integer
+        String[] commandWords = command.split("\\s"); // 0: "find", 1: keyword
+        if (logic.satisfiesMinimumDoneCommandLength(commandWords)) {
+            // this is a valid command
+            if (logic.determineIfValidDoneCommand(commandWords, this.sizeOf())) {
+                // this valid command has a valid number, so, mark as done
+                return doTask(Integer.parseInt(commandWords[1]));
+            } else {
+                // invalid input: attempting to do a non-existent task
+                throw new InputUnclearException("");
+            }
+        } else {
+            // invalid input
+            throw new DoneException("");
+        }
+    }
+
+    /**
+     * Gives a String representation of the result of a find command and finds the Tasks which contain the keyword.
+     *
+     * @param command the full, unedited command entered by the user.
+     * @return String representation of the result of finding Tasks with given keyword(s).
+     * @throws EmptyTaskListException an Exception given when TaskList is empty.
+     */
+    private String findTask(String command) throws EmptyTaskListException {
```

Comment:
these methods could possibly be abstracted out to another class for executing the commands